### PR TITLE
Fix unpacking zip package treats "../" as the top_level_directory

### PR DESCRIPTION
**Pull request by brycehuang30**
## Why are these changes needed?

TL;DR:
Parent path in package shouldn't be treated as top level directory.

The files in zip packages can contain relative paths like "..", and the current implementation of `get_top_level_dir_from_compressed_package()` extract the top_level_directory as "..", which should not be treated as a valid directory. When the top_level_directory is "..", `remove_dir_from_filepaths()` will break because it concatenates the top_level_directory after a temporary folder (see code snippet below, rdir is the top_level_directory), and the resulting path escapes the temporary directory (for example, if the temporary directory is `/tmp/tmp_abc`, the concatenated path becomes `/tmp/tmp_abc/..`).

https://github.com/ray-project/ray/blob/e171ad17082cc937b8c279790bd70cdeb22914a5/python/ray/_private/runtime_env/packaging.py#L901-L904

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by GokuMohandas** [comment]
@jjyao could you review this PR? Thank you!
